iothub/device/src/Pipeline/RetryDelegatingHandler.cs

```diff
@@ -14,6 +14,12 @@
 
 namespace Microsoft.Azure.Devices.Client.Transport
 {
+    internal enum ClientTransportStatus
```

Comment:
Can we put this is its own file, unless it should be part of the class only (in which case it'd also be private)?

Reply:
It is an enum defined on the namespace, so unfortunately we cannot make it private. I did place it within this file since this is the only place it is used. Would you prefer I move it to its own file?